phpunit/functional/RuleTicketTest.php

```diff
@@ -3403,6 +3403,41 @@ public function testAssignLocationFromUser(): void
         }
     }
 
+    /**
+     * Test that the PATTERN_DOES_NOT_EXISTS condition correctly matches
+     * when testing a rule with an empty input value.
+     *
+     * @return void
+     */
+    public function testPatternDoesNotExistsWithEmptyInput(): void
+    {
+        $rule = $this->createItem('RuleTicket', [
+            'name'        => 'test pattern does not exists',
+            'match'       => 'AND',
+            'is_active'   => 1,
+            'sub_type'    => 'RuleTicket',
+            'condition'   => \RuleTicket::ONADD,
+            'entities_id' => 0,
+        ]);
+
+        $this->createItem('RuleCriteria', [
+            'rules_id'  => $rule->getID(),
+            'criteria'  => 'locations_id',
+            'condition' => \Rule::PATTERN_DOES_NOT_EXISTS,
+            'pattern'   => 1,
+        ]);
+
+        $ruleticket = new \RuleTicket();
+        $ruleticket->getRuleWithCriteriasAndActions($rule->getID(), true, false);
+
+        $input = ['locations_id' => ''];
+        $check_results = [];
+        $ruleticket->testCriterias($input, $check_results);
+
+        $this->assertCount(1, $check_results);
+        $this->assertEquals(1, $check_results[array_key_first($check_results)]['result']);
```

Comment:
What is the meaning of this `1` value? I am not sure to understand what is tested here.

Reply:
Maybe checking entire array returned would be more easy to understand?